diff -up libmodbus-3.1.11/tests/unit-tests.sh._orig libmodbus-3.1.11/tests/unit-tests.sh
--- libmodbus-3.1.11/tests/unit-tests.sh._orig	2024-10-22 11:10:05.000000000 +0200
+++ libmodbus-3.1.11/tests/unit-tests.sh	2025-01-06 12:32:54.139591645 +0100
@@ -7,6 +7,7 @@ rm -f $client_log $server_log
 
 echo "Starting server"
 ./unit-test-server > $server_log 2>&1 &
+serverpid="$!"
 
 sleep 1
 
@@ -14,6 +15,6 @@ echo "Starting client"
 ./unit-test-client > $client_log 2>&1
 rc=$?
 
-killall unit-test-server
+kill "$serverpid"
 exit $rc
 
